src/retinanet.py

```diff
@@ -228,20 +236,32 @@ def forward(self, output, target):
         masked_loc_targets = loc_targets[mask].view(-1,4)  # [#pos,4]
 
         pos_neg = cls_targets > -1  # exclude ignored anchors
+        num_pos_neg = pos_neg.data.long().sum()
+
         mask = pos_neg.unsqueeze(2).expand_as(cls_preds)
         masked_cls_preds = cls_preds[mask].view(-1,self.num_classes)
 
         loc_loss = F.smooth_l1_loss(masked_loc_preds, masked_loc_targets, size_average=False)
         cls_loss = self.focal_loss(masked_cls_preds, cls_targets[pos_neg])
-        loss = (loc_loss+cls_loss)/num_pos
+
+        if num_pos_neg > 0:
+            cls_loss = cls_loss / num_pos_neg
+            loss = cls_loss * 10
```

Comment:
@taraspiotr maybe having fixed the batch norm problem we can drop the magic `10` .
If now please parametrize it as loc_cls_weight or something so that we won't forget about it

Reply:
@jakubczakon I think I can drop it for the time being, but I'm still wondering what is the best way to calculate final loss, but maybe it doesn't even matter.